suresh/array-str-methods.js

```diff
@@ -0,0 +1,97 @@
+var array = ['a', 1, 2, 'b', 'xyz'];
+
+function indexOf(array, ele) {
+  for (var i = 0; i < array.length; i++) {
+    if (array[i] === ele) {
+      return i;
+    } 
+  }
+  return "element is not present"
+}
+
+console.log('Array - indexOf: ' + indexOf(array, 'b'));
+
+var str = 'learning coding'
+
+console.log('String - indexOf: ' + indexOf(str, 'a'));
+
+var str = 'I%am%Learning%coding';
+
+function split(string, delimiter) {
+  var newArray = [];
+  var index = 0;
+  var spacing = [];
+  var wordIndex = 0;
+  
+  for (var i = 0; i < string.length; i++) {
+    if (string[i] === delimiter) {
+      spacing[index] = i;
+      index++
+    }
+  }
+  
+  console.log('spacing', spacing);
+  var startIndex = 0;
+  
+  for (var j = 0; j <= spacing.length; j++) {
+    var endIndex;
+    var word = '';
+    if (endIndex === string.length - 1) {
+      break;
+    }
+    endIndex = j == spacing.length ? string.length : spacing[j];
+
+    for (var k = startIndex; k < endIndex; k++) {
+      word = word + string[k];
+    }
+    startIndex = endIndex + 1;
+    newArray[wordIndex] = word;
+    wordIndex++;
+  }
+  return newArray;
+}
+
+console.log(split(str, '%'));
+
+var pets = ['dog', 'cat', 'rabbit', 'chickens', 'fish'];
+
+function slice(input, start, end) {
+  var slicedArray = [];
+ 
+  for (var i = start; i < input.length-1; i++) {
+  	if (i <= end) {
```

Comment:
This would work if the end is larger than the initial array but it would save time if you didnt have to iterate. If the end is greater than the array, just return.